### PR TITLE
FEATURE / Integrate function groups share permissions + script for testing purposes

### PR DESCRIPTION
**Pull request by XaviPeiro**
…mail'

## Proposed Changes

- Integrate/add "share function group".
- Integrate/add "create function group".
- Client (AiEngine) is not assuming as error anything diff than http.status_code=200,  but anything diff than http.status_code=2** 
- Provide a script to create a function group and share it (mainly for testing purposes in staging).
## Linked Issues
None

## Types of changes

_What type of change does this pull request make (put an `x` in the boxes that apply)?_

- [ ] Bug fix (non-breaking change that fixes an issue).
- [x] New feature added (non-breaking change that adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected).
- [ ] Documentation update.
- [ ] Something else (e.g., tests, scripts, example, deployment, infrastructure).

## Checklist

_Put an `x` in the boxes that apply:_

  - [ ] I have read the [CONTRIBUTING](/CONTRIBUTING.md) guide
  - [ ] Checks and tests pass locally

### If applicable

  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] I have checked that code coverage does not decrease
  - [ ] I have added/updated the documentation

## Further comments

_[if this is a relatively large or complex change, kick off a discussion by explaining why you chose the solution you did, what alternatives you considered, etc...]_

### CODE REVIEW

**Review by qati — CHANGES_REQUESTED** [review]
Please make the functionality part of the sdk itself.

Create a scripts folder and using the sdk functions make a small example how to use it. 
Document it.
Don't commit JWT token.

File: ai_engine_sdk/client.py [hunk 1]
```diff
@@ -61,6 +62,11 @@ class FunctionGroup(BaseModel):
     isPrivate: bool
 
 
+class CreateFunctionGroupSchema(BaseModel):
+    name: str
+    isPrivate: bool
```
Comment (NindoK) [inline ai_engine_sdk/client.py:67]: is_private? Just coz python :D

**Review by NindoK — APPROVED** [review]

File: ai_engine_sdk/client.py [hunk 2]
```diff
@@ -289,22 +300,35 @@ async def get_private_function_groups(self) -> List[FunctionGroup]:
...
-            totalCredits=response['total_credit'],
-            usedCredits=response['used_credit'],
-            availableCredits=response['available_credit']
+    async def create_function_group(
+            self,
+            is_private: bool,
+            name: str
+    ) -> FunctionGroup:
+        payload = {
+            "isPrivate": is_private,
```
Comment (qati) [inline ai_engine_sdk/client.py:316]: this is a good point here. Currently only private FG-s can be created, the api won't allow public one.

File: use_cases/py_scripts/create_function_group_and_share.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import asyncio
```
Comment (qati) [inline use_cases/py_scripts/create_function_group_and_share.py:1]: can this be moved to examples/create_function_group_and_share.py? Also if we could support sharing existing FG would be nice.

**Review by qati — CHANGES_REQUESTED** [review]
Can we add a short documentation to the readme about the new methods? And link the examples

File: ai_engine_sdk/client.py (hunk 2 above)
Comment (XaviPeiro) [reply ai_engine_sdk/client.py:316]: Tbh, I did not notice it... 
There is any business/logic reason to restrict other than internal ppl to create public function groups? 
image

File: ai_engine_sdk/client.py (hunk 1 above)
Comment (XaviPeiro) [reply ai_engine_sdk/client.py:67]: Cannot, we should migrate to pydanticv2 and before that adding tests. Nonetheless, task created for all these things.

**Review by OKEAMAH — APPROVED** [review]
Good

**Review by qati — APPROVED** [review]

**Pull request merged**